android-components/components/feature/addons/src/test/java/AddonManagerTest.kt

```diff
@@ -70,11 +70,20 @@ class AddonManagerTest {
     @Test
     fun `getAddons - queries addons from provider and updates installation state`() = runTestOnMain {
         // Prepare addons provider
+        // addon1 (ext1) is a featured extension that is already installed.
         val addon1 = Addon(id = "ext1")
+        // addon2 (ext2) is a featured extension that is not installed.
         val addon2 = Addon(id = "ext2")
+        // addon3 (ext3) is a featured extension that is marked as disabled.
         val addon3 = Addon(id = "ext3")
+        // addon4 (ext4) and addon5 (ext5) are not featured extensions but they are installed.
+        val addon4 = Addon(id = "ext4")
+        val addon5 = Addon(id = "ext5")
         val addonsProvider: AddonsProvider = mock()
         whenever(addonsProvider.getFeaturedAddons(anyBoolean(), eq(null), language = anyString())).thenReturn(listOf(addon1, addon2, addon3))
+        // addon4 (ext4) and addon5 (ext5) should already be installed, which is why we expect a call
+        // to `getAddonsByGUIDs()` with a list of two GUIDs.
+        whenever(addonsProvider.getAddonsByGUIDs(eq(listOf("ext5", "ext4")), anyBoolean(), eq(null), language = anyString())).thenReturn(listOf(addon4, addon5))
```

Comment:
I am honestly not sure why `getAddonsByGUIDs()` receives a list with `"ext5"` first. I would have expected `"ext4"` first, and then `"ext5"`.

Reply:
That's really odd!